spec/calculate_spec.rb

```diff
@@ -17,58 +16,55 @@
 
       expect(Calculate.format_date).to eql(today_date_formatted)
     end
-    it 'if date passed in, return passed in value', :focus do
-      expect(Calculate.format_date("123456")).to eql("123456")
-    end
-    it 'if date passed in, raises error if length does not equal 6' do
-      expect{Calculate.format_date("31420")}.to raise_error(ArgumentError)
-    end
-    it 'if date passed in, raises error if contains characters besides numbers' do
-      expect{Calculate.format_date("453f34")}.to raise_error(ArgumentError)
-    end
+
+      context 'when date passed in' do
+
+        it 'return passed in value'  do
+          expect(Calculate.format_date("123456")).to eql("123456")
+        end
+
+        context 'when length does not equal 6' do
+          it 'raises error' do
+            expect{Calculate.format_date("31420")}.to raise_error(ArgumentError)
+          end
+        end
+
+        context 'when containing characters besides numbers' do
+          it 'raises error ' do
+            expect{Calculate.format_date("453f34")}.to raise_error(ArgumentError)
+          end
+        end
+      end
   end
 
   describe '::offsets' do
-
     it 'returns hash containing last four digits of date squared' do
-      formatted_date = '011519'
-
-      expect(Calculate.offsets(formatted_date)).to include({a_offset:7,b_offset:3,c_offset:6,d_offset:1 })
+      expect(Calculate.offsets('011519')).to include({a_offset:7,b_offset:3,c_offset:6,d_offset:1 })
     end
     it 'returns hash containing last four digits of date squared test 2' do
-      formatted_date = '011517'
-
-      expect(Calculate.offsets(formatted_date)).to include({a_offset:1,b_offset:2,c_offset:8,d_offset:9 })
+      expect(Calculate.offsets('011517')).to include({a_offset:1,b_offset:2,c_offset:8,d_offset:9 })
     end
-
   end
 
   describe '::keys' do
     it 'splits random 5 digit number into 4 two-digit hash entries' do
-      random_number = "02715"
-
-      expect(Calculate.keys(random_number)).to include({a_key:'02',b_key:'27',c_key:'71',d_key:'15' })
+      expect(Calculate.keys("02715")).to include({a_key:'02',b_key:'27',c_key:'71',d_key:'15' })
     end
 
     it 'splits random 5 digit number into 4 two-digit hash entries test 2' do
-      random_number = "00813"
-
-      expect(Calculate.keys(random_number)).to include({a_key:'00',b_key:'08',c_key:'81',d_key:'13' })
+      expect(Calculate.keys("00813")).to include({a_key:'00',b_key:'08',c_key:'81',d_key:'13' })
     end
   end
 
   describe '::random_number_string' do
-
-    it 'should be type string' do
-      expect(Calculate.random_number_string).to be_a(String)
-    end
-
-    it 'generates string with length of 5' do
-      expect(Calculate.random_number_string.length).to be(5)
-    end
-
-    it 'generates random number string with leading zeros if neccessary thats equal to 5 digits in length' do
-      expect(Calculate.random_number_string.length).to be(5)
+    
+    context 'when number is generated' do
```

Comment:
The way I would test this would be by injecting an anonymous stub function:

```
fake_random = \_ -> "a"
```

That's Elm'ish syntax. Ruby has anonymous functions though (I think they are called `lambda`s). Same basic idea though. We have a function that will return one element which we throw away. . . and instead just return `"a"`.

Then the test could just be:

```
result = Calculate.some_func(fake_random)

expect(result).to be("aaaaa")
```